Source/WebCore/dom/wpe/PointerEventWPE.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2023 Pengutronix e.K.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "PointerEvent.h"
+
+#if ENABLE(TOUCH_EVENTS) && PLATFORM(WPE)
+
+#include "EventNames.h"
+
+namespace WebCore {
+
+Ref<PointerEvent> PointerEvent::create(const AtomString& type, Ref<WindowProxy>&& view, const Touch& touch, IsPrimary isPrimary)
+{
+    // FIXME: Pressure is not supported currently. In this case the pressure should be set to 0.5 when in
+    // 'active buttons state' and to 0 otherwise according the specification:
```

Comment:
No, as you can see, we don't pass an actual pressure value here, but rather use 0 and 0.5 as specified for platforms that do not support pressure:
> For hardware and platforms that do not support pressure, the value MUST be 0.5 when in the [active buttons state](https://www.w3.org/TR/pointerevents/#dfn-active-buttons-state) and 0 otherwise. Note: all pointerup events will have pressure 0.

Source: https://www.w3.org/TR/pointerevents/#dom-pointerevent-pressure

Actual pressure handling can be added later. Most touch controllers and drivers do not have this information anyway.

Reply:
This is now reworked completely, please have a look at the recent version.